File: src/main/java/io/pageobject/generator/locator/protractor/NgRepeatLocatorSource.java
package io.pageobject.generator.locator.protractor;

import io.pageobject.generator.GeneratorContext;
import io.pageobject.generator.locator.AbstractLocatorSource;
import org.jsoup.nodes.Element;

import static io.pageobject.generator.Expressions.allowIndexExpressionOrNoExpressions;
import static io.pageobject.generator.Expressions.extractExactRepeater;
import static io.pageobject.generator.attribute.Attributes.getRepeaterAttribute;

public class NgRepeatLocatorSource extends AbstractLocatorSource {

    @Override
    public String extractLocatorValue(Element element, GeneratorContext context) {
        return allowIndexExpressionOrNoExpressions(extractExactRepeater(getRepeaterAttribute(element)));
    }

    public String locatorTemplate() {
        return "by.exactRepeater('%s')";
    }

}

File: lib/locator/protractor/ng-repeat-locator-source.js
"use strict";

const AbstractLocatorSource = require('../abstract-locator-source');
const _ = require('lodash');
const allowIndexExpressionOrNoExpressions = require('../../expressions').allowIndexExpressionOrNoExpressions;
const extractExactRepeater = require('../../expressions').extractExactRepeater;
const getRepeaterAttribute = require('../../attribute/attributes').getRepeaterAttribute;

const template = _.template("by.exactRepeater('<%= value %>')");

class NgRepeatLocatorSource extends AbstractLocatorSource {

  extractLocatorValue(context) {
    return allowIndexExpressionOrNoExpressions(extractExactRepeater(getRepeaterAttribute(context.element)));
  }

  locatorTemplate() {
    return template;
  }

}

module.exports = NgRepeatLocatorSource;